instrumentation/racecar/lib/opentelemetry/instrumentation/racecar/instrumentation.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry-instrumentation-rdkafka'
+
+module OpenTelemetry
+  module Instrumentation
+    module Racecar
+      # The Instrumentation class contains logic to detect and install the Racecar instrumentation
+      class Instrumentation < OpenTelemetry::Instrumentation::Base
+        MINIMUM_VERSION = Gem::Version.new('2.0')
+
+        compatible do
+          Gem::Version.new(::Racecar::VERSION) >= MINIMUM_VERSION
+        end
+
+        install do |_config|
+          OpenTelemetry::Instrumentation::Rdkafka::Instrumentation.instance.install({})
```

Comment:
I'm a bit confused here. So is the idea that the racecar instrumentation depends on the rdkafka instrumentation?

1. Is it possible to check if the instrumentation is already installed before attempting to re-install it?
2. What's the desired behavior if the rdkafka instrumentation has been intentionally disabled by the user via env var? Are we essentially overwriting that choice.
2a. Hypothetical since there's no config options at the moment, but won't we automatically overwriting any configuration the user has supplied for their `rdkafka` instrumentation should it be aded in the future?

I guess i have to think a bit more closely about what the "right" behavior should be, but imo we should be respecting if this instrumentation has been disabled and, imo, its not clear to me why we need to take rdkafka as a dependancy.

Reply:
Yes, my intention here was to rely on the Rdkafka implementation of producer spans as adding a Racecar implementation would duplicate them like this:

```
racecar send span:
  rdkafka send span:
    consumer span
```

There is a similar dependency between the Rack instrumentation with Rails and Sinatra so I did assume it would be alright to follow that pattern, but it is being used slightly different here and less extensively.

> 1. Is it possible to check if the instrumentation is already installed before attempting to re-install it?

Yes, there is an interface we could use.

> 2. What's the desired behavior if the rdkafka instrumentation has been intentionally disabled by the user via env var? Are we essentially overwriting that choice.
>    2a. Hypothetical since there's no config options at the moment, but won't we automatically overwriting any configuration the user has supplied for their `rdkafka` instrumentation should it be aded in the future?

Yes, I agree, there probably needs to be some more consideration here. If we don't want to load rdkafka's instrumentation then we should probably load some instrumentation here for producer spans.

As alternatives to this, I can think of the following options:
- Duplicate the spans (there is precedence for this with Faraday and it wrapping of other clients like net/http)
- Have the racecar producer instrumentation used `untraced` so that the rdkafka producer spans are dropped

What do you think?